_posts/2022-03-07-using-private-image-repositories.markdown

```diff
@@ -0,0 +1,201 @@
+
+1. Login to quay.io:
+```bash
+podman login quay.io
+```
+You will be prompted for your username and password:
+```bash
+$ podman login quay.io                                                                                                     
+Username: xxxxxx
+Password: ******
+Login Succeeded!
+```
+
+1. Pull `docker.io/nginx:1.14.2`
+```bash
+podman pull docker.io/nginx:1.14.2
+```
+
+1. Tag `docker.io/nginx:1.14.2` with a private name
+```bash
+podman tag docker.io/nginx:1.14.2 quay.io/<your quay.io username>/nginx:1.14.2
+```
+
+1. Push `nginx:1.14.2` to quay.io
+```bash
+podman push quay.io/<your quay.io username>/nginx:1.14.2
+```
+
+### Deployment
+
+#### Sanity check
+Before we start using private repository feature of Flotta, let's see what happens when you try to deploy private image in a simple `EdgeDeployment`:
+1. Confirm that you have an `EdgeDevice` labelled with `dc=home`
+```bash
+$ kubectl get edgedevice -l dc=home
+NAME                                          AGE
+fedora-54b9dd17-9971-11ec-bcee-7085c256610d   149m
+```
+
+1. Create the `EdgeDeployment` that will be deployed to each `EdgeDevice` with `dc=home` label
+```shell
+kubectl apply -f -<<EOF
+apiVersion: management.project-flotta.io/v1alpha1
+kind: EdgeDeployment
+metadata:
+  name: nginx-deployment
+spec:
+  type: pod
+  deviceSelector:
+    matchLabels:
+      dc: home
+  pod:
+    spec:
+      containers:
+        - name: nginx
+          image: quay.io/jdzon/nginx:1.14.2
+          ports:
+            - containerPort: 80
+              hostPort: 9090
+EOF
+```
+
+1. Confirm that `nginx-deployment` pod wouldn't start on the edge device by checking deployment phase (not `Running`) and events reported for your `EdgeDevice` object: 
+```shell
+$ kubectl describe edgedevice
+Name:         fedora-54b9dd17-9971-11ec-bcee-7085c256610d
+Namespace:    default
+Labels:       dc=home
+              device.cpu-architecture=x86_64
+...
+Status:
+  Deployments:
+    Last Transition Time:  2022-03-07T10:35:53Z
+    Name:                  nginx-deployment
+    Phase:                 Created
+...
+Events:
+  Type     Reason  Age                  From                       Message
+  ----     ------  ----                 ----                       -------
+  Warning  Failed  4m10s (x2 over 16m)  edgedeployment-controller  error playing YAML file: initializing source docker://quay.io/jdzon/nginx:1.14.2: reading manifest 1.14.2 in quay.io/jdzon/nginx: unauthorized: access to the requested resource is not authorized
+```
+
+#### Creating auth file secret
+To use private container image, user needs to obtain credentials for their image repository - contents of the auth file generated when one executes
+```bash
+podman login quay.io
+```
+
+Usually that would be `$HOME/.docker/config.json` or `${XDG_RUNTIME_DIR}/containers/auth.json` file.
+
+When you have the auth file, you need to create a kubernetes secret containing it under `.dockerconfigjson` key:
```

Comment:
Add a link to https://kubernetes.io/docs/concepts/configuration/secret/ makes sense here?